### PR TITLE
[canvas-ity] New port

### LINKED ISSUE #33617: [New Port Request] canvas_ity
### Library name

canvas_ity

### Library description

A simple C++ drawing library with HTML5 <canvas>-like API

### Source repository URL

https://github.com/a-e-k/canvas_ity

### Project homepage (if different from the source repository)

_No response_

### Anything else that is useful to know when adding (such as optional features the library may have that should be included)

This library is pure header only.

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

Close #33617

### CODE REVIEW

**Comment by Adela0814** [comment]
No feature needs to test.
Usage tested pass on `x64-windows`.

**Review by Adela0814 — DISMISSED** [review]

File: ports/canvas-ity/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO a-e-k/canvas_ity
+    REF f32fbb37e2fe7c0fcaee6ebdc02d3e5385603fd5
+    SHA512 37111c445ce36705f43ebd9c5acd68fc7dd1ddaf9ffd1b857a936ff7c0f861fdad26ad5af5c841f72898f561447e5efeb70814e928b961ecb68ac6ae39cae5b9
+    HEAD_REF master
+)
+
+file(INSTALL "${SOURCE_PATH}/src/canvas_ity.hpp" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
+
+vcpkg_install_copyright(FILE_LIST "${CMAKE_CURRENT_LIST_DIR}/LICENSE")
```
Comment (BillyONeal) [inline ports/canvas-ity/portfile.cmake:13]: I'm concerned about the license not coming from the source code, meaning that on update if the license changes, we might not notice and be telling people the wrong thing. Related: https://github.com/microsoft/vcpkg/pull/28644

Comment (xiaozhuai) [reply ports/canvas-ity/portfile.cmake:13]: On the vacation now, would you please change that for me？  😄

Comment (xiaozhuai) [reply ports/canvas-ity/portfile.cmake:13]: @BillyONeal Hi there, how should I change this PR?

**Comment by xiaozhuai** [comment]
@BillyONeal Hi there, I've update this pr.

File: ports/canvas-ity/portfile.cmake (hunk 1 above)
Comment (Adela0814) [reply ports/canvas-ity/portfile.cmake:17]: @xiaozhuai Please refer to https://github.com/microsoft/vcpkg/pull/32206/commits/109fdce4b7632f637e2051ab1ac552ba48588056.

Comment (xiaozhuai) [reply ports/canvas-ity/portfile.cmake:17]: @Adela0814 @BillyONeal 
I don't think https://github.com/microsoft/vcpkg/commit/109fdce4b7632f637e2051ab1ac552ba48588056 will suit well for the case here.
`canvas-ity` is under the ISC license. The license content are in the code file.
So we can check it in `canvas_ity.hpp`.

If we refer to https://github.com/microsoft/vcpkg/commit/109fdce4b7632f637e2051ab1ac552ba48588056, then we may miss this in the future when the license changes.

Comment (dg0yt) [reply ports/canvas-ity/portfile.cmake:17]: Read the header, remove everything starting at `/ ======== ABOUT ========`, write `copyright`?

**Comment by Adela0814** [comment]
I noticed that the upstream is adding the license to the source code, so let's wait for the PR https://github.com/a-e-k/canvas_ity/pull/3 to merge.

**Comment by xiaozhuai** [comment]
> I noticed that the upstream is adding the license to the source code

@Adela0814 
This PR is about a year ago. It doesn't look like upstream is interested in merging it.

**Comment by Adela0814** [comment]
@BillyONeal Please review this PR again. It seems that the upstream has not been active for nearly a year.

**Review by vicroms — DISMISSED** [review]
Please make the change suggested by @dg0yt to obtain the license text from a header file.

**Comment by xiaozhuai** [comment]
@vicroms @dg0yt 
I decide to close this PR. The upstream never accept PR and fix issue and has not been active for a year.
Perhaps not add this port to vcpkg is right choice.

**Comment by a-e-k** [comment]
I'm still here; I've just been busy with life and a bit of crunch mode at work.

I've now [added](https://github.com/a-e-k/canvas_ity/commit/fc5c115d8ca8be0cf81c2247270cc955f78b6138) a standalone `LICENSE.txt` to the repo.

**Comment by xiaozhuai** [comment]
> I've now [added](https://github.com/a-e-k/canvas_ity/commit/fc5c115d8ca8be0cf81c2247270cc955f78b6138) a standalone LICENSE.txt to the repo.

@a-e-k 
Thanks! I've reopen this PR.

@BillyONeal @vicroms @Adela0814 
Please review this pr again.

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**